src/omlt/linear_tree/lt_definition.py

```diff
@@ -0,0 +1,408 @@
+                )
+                ubs = scaling_object.get_scaled_input_expressions(
+                    {k: t[1] for k, t in unscaled_input_bounds.items()}
+                )
+
+                scaled_input_bounds = {
+                    k: (lbs[k], ubs[k]) for k in unscaled_input_bounds.keys()
+                }
+
+            # If unscaled input bounds provided and no scaler provided, scaled
+            # input bounds = unscaled input bounds
+            elif unscaled_input_bounds is not None and scaling_object is None:
+                scaled_input_bounds = unscaled_input_bounds
+            elif unscaled_input_bounds is None:
+                raise ValueError(
+                    "Input Bounds needed to represent linear trees as MIPs"
+                )
+
+        self.__unscaled_input_bounds = unscaled_input_bounds
+        self.__scaled_input_bounds = scaled_input_bounds
+
+        self.__splits, self.__leaves, self.__thresholds = _parse_tree_data(
+            lt_regressor, scaled_input_bounds
+        )
+
+        self.__n_inputs = _find_n_inputs(self.__leaves)
+        self.__n_outputs = 1
+
+    @property
+    def scaling_object(self):
+        """Returns scaling object"""
+        return self.__scaling_object
+
+    @property
+    def scaled_input_bounds(self):
+        """Returns dict containing scaled input bounds"""
+        return self.__scaled_input_bounds
+
+    @property
+    def splits(self):
+        """Returns dict containing split information"""
+        return self.__splits
+
+    @property
+    def leaves(self):
+        """Returns duct containing leaf information"""
+        return self.__leaves
+
+    @property
+    def thresholds(self):
+        """Returns dict containing threshold information"""
+        return self.__thresholds
+
+    @property
+    def n_inputs(self):
+        """Returns number of inputs to the linear tree"""
+        return self.__n_inputs
+
+    @property
+    def n_outputs(self):
+        """Returns number of outputs to the linear tree"""
+        return self.__n_outputs
+
+
+def _find_all_children_splits(split, splits_dict):
+    """
+    This helper function finds all multigeneration children splits for an
+    argument split.
+
+    Arguments:
+        split --The split for which you are trying to find children splits
+        splits_dict -- A dictionary of all the splits in the tree
+
+    Returns:
+        A list containing the Node IDs of all children splits
+    """
+    all_splits = []
+
+    # Check if the immediate left child of the argument split is also a split.
+    # If so append to the list then use recursion to generate the remainder
+    left_child = splits_dict[split]["children"][0]
+    if left_child in splits_dict:
+        all_splits.append(left_child)
+        all_splits.extend(_find_all_children_splits(left_child, splits_dict))
+
+    # Same as above but with right child
+    right_child = splits_dict[split]["children"][1]
+    if right_child in splits_dict:
```

Comment:
The lines below were not covered. Seems odd. Do we need a different test?

Reply:
Never ran due to the structure of the trained tree. I added another test that ensures these children finders work regardless of tree structure.